sdk/eventhub/Azure.Messaging.EventHubs.Processor/perf/Scenarios/ProcessEventTest.cs

```diff
@@ -82,21 +102,48 @@ private async Task ProcessEventAsync(ProcessEventArgs args)
             }
         }
 
+        /// <summary>
+        ///  Performs the tasks needed to report an error observed as part of the
+        ///  processor's operation.
+        /// </summary>
+        ///
+        /// <param name="arg">The <see cref="ProcessErrorEventArgs"/> instance containing the error information.</param>
+        ///
         private Task ProcessErrorAsync(ProcessErrorEventArgs arg)
         {
             ErrorRaised(arg.Exception);
             return Task.CompletedTask;
         }
 
+        /// <summary>
+        ///   Performs the tasks needed to initialize and set up the environment for the test scenario.
+        ///   This setup will take place once for each instance, running after the global setup has
+        ///   completed.
+        /// </summary>
+        ///
         public override async Task SetupAsync()
         {
             await base.SetupAsync();
+
+            EventProcessorClient.ProcessEventAsync += ProcessEventAsync;
```

Comment:
I don't feel too strongly either way, but the typical pattern in perf tests is to do things in ctor/dispose if possible, and only use setup/cleanup if required (e.g. calling async methods).   So I think attach/detach event handlers could live in ctor/dispose.  But if you think the code is cleaner having everything in setup/cleanup, that's fine too.  It's just a convention we've been using.

Reply:
We need it in Setup because the client doesn't exist in the constructor.   We have to grab consumer group from the transient scope in the base class.   The scope is an async operation, so the client isn't created until base.Setup.

Reply:
We could assume the default consumer group, but generally, we want to allow the scope to dictate since it owns the set of resources and may have been asked to use specific consumer groups.